celltype_annotation/analysis/singler-rms-comparison.Rmd

```diff
@@ -0,0 +1,245 @@
+labels <- c("label.main", "label.fine", "label.ont")
+celltype_assignment_list <- purrr::map(labels,
+                                       \(label_type) celltype_assignments_table(coldata_df,
+                                                                                label_type)) |>
+  purrr::set_names(labels)
+```
+
+
+```{r, fig.width=10, fig.height=5}
+# print out barcharts
+purrr::imap(celltype_assignment_list, 
+           \(celltype_assignments, label_type) celltype_barchart(celltype_assignments,
+                                                                 label_type))
+```
+
+```{r}
+# print out tables 
+purrr::map(celltype_assignment_list, 
+            \(celltype_assignments) 
+            celltype_table(celltype_assignments))
+```
+
+## Comparison of SingleR annotations to submitter annotations
+
+This section will directly compare cell type annotations using `HumanPrimaryCellAtlasData` or `BlueprintEncodeData` with `SingleR` vs. manual annotations. 
+Each chunk displays a heatmap directly comparing a `SingleR` reference (columns) to manual annotations (rows).
+
+```{r}
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$label.ont_BlueprintEncodeData)
+```
+```{r}
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$label.main_BlueprintEncodeData)
+```
+```{r, fig.height=7, fig.width=7}
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$label.ont_HumanPrimaryCellAtlasData)
+```
+
+```{r, fig.height=5, fig.width=7}
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$label.main_HumanPrimaryCellAtlasData)
+```
+
+## Median delta score 
+
+The last thing we will look at is the distribution of the delta between the top and median scores (referred to as the delta median here).
+We can grab the complete `SingleR` results from the `metadata` of the annotated SCE object.
+These results contain a matrix of all the scores for each cell and for each label. 
+We expect that a reference that is more appropriate will have higher delta medians. 
+
+```{r warning=FALSE, message=FALSE}
+# first just grab the singleR objects so we can directly use those as input to the plotScoreHeatmap function
+singler_results <- metadata(singler_annotated_sce)$singler_results
+
+median_delta_results <- singler_results |> 
+  purrr::map(get_median_delta) |>
+  dplyr::bind_rows(.id = "reference") |>
+  tidyr::separate(reference, into = c("label_type", "reference"), sep = "_", extra = "merge") |>
+  # classify each label as pruned or not
+  dplyr::mutate(pruned = ifelse(is.na(pruned.labels), "Yes", "No"))
+
+# plot the distribution of median_delta for each reference, faceting by label type (main, fine, ont)
+plot_metric_distribution(median_delta_results, "median_delta")
```

Comment:
I think this plot is a little tight and tough to see, would putting legend on the bottom help? again i didn't test this, but maybe it helps!
```suggestion
plot_metric_distribution(median_delta_results, "median_delta") + 
  theme(legend.position = "bottom")
```